src/js/menu.js

```diff
@@ -110,18 +164,39 @@ var showSensorsMenu = function(thermostat) {
       items: menuItems
     }]
   });
-  menu.show();
+  
+  hvacModeMenu.on('select', function(e) {
+    var postRequest = Utils.createChangeModeRequest(thermostat, e.item.value);
+    ecobeeApi.postThermostat(postRequest, 
+      function() {
+        hvacModeMenu.hide();
+        if (menu) menu.hide();
+      }, 
+      function(error) {
+        console.log('error setting hvac mode: '+error);
+    });
+  });
+  
+  hvacModeMenu.show();
 };
 
 this.exports = {
-  show: function(thermostat) {
-    if (menu) {
-      menu.hide();
-      menu = null;
+  show: function(thermostatList) {   
+    var thermostat;
+    var selectedThermostatId = Settings.data('selectedThermostatId');
+    if(selectedThermostatId){
+      thermostat = Utils.selectThermostat(selectedThermostatId,thermostatList);
+    }
+    else{
+      thermostat = thermostatList[0];
     }
+
     var menuItems = [];
     var hasHold = Utils.hasHold(thermostat);
     var hasSensors = Utils.hasSensors(thermostat);
+    if(thermostatList.length > 1){
+      menuItems.push({title: 'Thermostats'});
```

Comment:
Let's move closer to the bottom. Switching between thermostats may not happen more often then changing holds. How often do you switch between tstats by the way? 
I think order should be: 
- Sensors
- Home and Hold
- Away and Hold
- Sleep and Hold
- Thermostats
- Change Mode

Also we already have too many items for such a small screen. We might want to have thermostat selection and mode changes in sub-menu called maybe "Settings".